### PR TITLE
TRC-24 - chore: Add task running and linting

### PR DESCRIPTION
**Pull request by wijohnst**
# Adds shared typescript configuration files, `husky`, `eslint`, and `prettier`

## TL/DR
This PR add several DX and build tooling improvements, mainly focused around shared typescript configuration and linting. It also adds the `husky` runner for orchestration of commit hooks.

## Overview of Changes
`.gitignore`
- Ads `.vscode` directory

`.husky/pre-commit`
- Adds pre-commit config

`.prettier.json`
- Adds global `prettier` config

`apps/api/eslintrc.js`
- Adds workspace `eslint` config

`apps/api/package.json`
- Adds `lint` script
- Adds `eslint` as a dev dependency

`apps/api/tsconfig.json`
- Refactors to use shared config from `tsconfig` package

`apps/web/eslintrc.js`
- Adds workspace `eslint` config

`apps/web/package.json`
- Refactors `lint` script
- Adds `eslint` as a dev dependency

`package.json`
- Adds `preinstall` script
- Adds `dev:server` script - will run the server without running other workspaces
- Adds `husky` and `lint-staged` as dev depndencies
- Adds `lint-staged` config

`packages/config/package.json` 
- Adds `config` workspace

`packages/config/eslint-base.js`
- Adds base `eslint` configuration - can be shared between workspaces

`eslint-react.js` 
- Adds `react` specific `eslint` config - can extend `eslint-base`

`node-express.json`
- Adds Node-specific typescript config

`pnpm-workspace.yaml`
- Refactors `apps` workspace path to wildcard

### CODE REVIEW

**Review by mmcbride2929 — APPROVED** [review]
looks good!

**Comment by wijohnst** [comment]
Closes #24
